### PR TITLE
Basic analysis

### PR DESCRIPTION
**Pull request by panord**
Made generic interface for analyses, 

Analyses using only intervals are called BasicAnalyses
Analysis is an extension of BasicAnalyses where it's intervals contain bounding boxes for that interval
AnalysisProxy(Meta) cleanup up and method name clarified.

### CODE REVIEW

File: ViAn/GUI/analysisslider.cpp [hunk 1]
```diff
@@ -76,9 +76,9 @@ void AnalysisSlider::set_analysis(AnalysisMeta* analysis) {
     repaint();
 }
 
-void AnalysisSlider::set_tag(Tag *tag)
+void AnalysisSlider::set_tag(BasicAnalysis *tag)
 {
-    set_analysis(new AnalysisMeta(static_cast<Analysis>(*tag)));
+    set_analysis(dynamic_cast<BasicAnalysis*>(tag));
```
Comment (fredrikiselius) [inline ViAn/GUI/analysisslider.cpp]: Isn't tag already a BasicAnalysis*?

File: ViAn/GUI/videowidget.cpp [hunk 2]
```diff
@@ -492,13 +492,14 @@ void VideoWidget::analysis_btn_clicked() {
 }
 
 void VideoWidget::tag_frame() {
-    if (m_tag->type == TAG){
-        if (m_tag->add_frame(current_frame)) {
-            emit tag_updated(m_tag);
+    Tag* tag = dynamic_cast<Tag*>(m_tag);
+    if (tag->getType() == TAG){
```
Comment (fredrikiselius) [inline ViAn/GUI/videowidget.cpp]: Isn't a tag always of type TAG?

File: ViAn/Project/Analysis/analysisinterval.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    return false;
+}
+
+bool AnalysisInterval::in_interval(int frame) const
+{
+   return frame >= m_start && frame <= m_end;
+}
+
+int AnalysisInterval::getStart() const
```
Comment (fredrikiselius) [inline ViAn/Project/Analysis/analysisinterval.cpp]: get_start

File: ViAn/Project/Analysis/analysisinterval.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+{
+    return m_start;
+}
+
+int AnalysisInterval::getEnd() const
```
Comment (fredrikiselius) [inline ViAn/Project/Analysis/analysisinterval.cpp]: get_end

File: ViAn/Project/Analysis/analysisinterval.cpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+{
+    return m_end;
+}
+
+std::pair<int, int> AnalysisInterval::getInterval()
```
Comment (fredrikiselius) [inline ViAn/Project/Analysis/analysisinterval.cpp]: get_interval

File: ViAn/Project/Analysis/basicanalysis.cpp [hunk 6]
```diff
@@ -56,17 +56,17 @@ void BasicAnalysis::write(QJsonObject &json){
...
 }
 
-int BasicAnalysis::getType() const
+int BasicAnalysis::get_type() const
 {
     return type;
 }
 
-interval_set BasicAnalysis::getIntervals() const
+interval_set BasicAnalysis::get_interval() const
```
Comment (henpe071) [inline ViAn/Project/Analysis/basicanalysis.cpp]: Should still be plural, intervals

**Review by fredrikiselius — APPROVED** [review]

**Pull request merged**